src/general/index.ts

```diff
@@ -110,3 +110,121 @@ export function getSelectedTabIndex(tabs: any[], selectedTabName: string): numbe
     return defaultIndex;
   }
 }
+
+
+/**
+ * This function can download a file on user's machine either directly by a url or a blob object.
+ *
+ * @param {
+ *  file:File | null;
+ *  type:string; fileName:string;
+ *  downloadMethod:string;
+ *  fileExtension:string;
+ *  fileUrl:string | null
+ * } downloadConfig
+ *
+ * @remarks
+ *
+ * Please ensure you are passing the appropriate downloadMethod type -
+ * <br />
+ * <br />
+ * <p>'url' method expects the fileUrl argument</p>
+ * <p>'blob' expects the file argument</p>
+ *
+ * <br />
+ * <h4>downloadConfig properties</h4>
+ * <ul>
+ *  <li> <span style="font-weight: bold;">file</span> => BlobObject or null. Required if downloadMethod is 'blob' </li>
+ *  <li> <span style="font-weight: bold;">type</span> => MIME-TYPE of the file. 'application/pdf', 'application/gzip', 'image/png' </li>
+ *  <li> <span style="font-weight: bold;">fileName</span> => Expected name of the downloaded file </li>
+ *  <li> <span style="font-weight: bold;">downloadMethod</span> => 'blob' or 'url' </li>
+ *  <li> <span style="font-weight: bold;">fileExtension</span> => Expected extension of the downloaded file </li>
+ *  <li> <span style="font-weight: bold;">fileUrl</span> => downloadable file's url. Required if downloadMethod is 'url' </li>
+ * </ul>
+ *
+ * @example
+ * ```
+ * downloadFile({
+ *  file: fileBlobObject,
+ *  type: 'application/pdf',
+ *  fileName: 'MyFile',
+ *  fileExtension: 'pdf',
+ *  downloadMethod: 'blob',
+ *  fileUrl: null
+ * }) // *Downloads file of type PDF on the client's machine named MyFile.pdf*
+ * ```
+ *
+ * @category General Method
+ */
+export function downloadFile(downloadConfig:{ file:File | null; type:string; fileName:string; downloadMethod:string; fileExtension:string; fileUrl:string | null }) {
```

Comment:
This method should be smaller. There should be 1 method to downloadFileFromUrl and another downloadFileFromBlob. Both should be separate functions and don't export these methods. You can even find something so that documentation dont get generated for these. THen write wrapper method which uses these 2 and export that method.

Reply:
It might lead to over abstraction.. These 2 methods serve no purpose separately. They will always be a part of the wrapper method. So it will not make a difference if they exist out of the method block or within. I believe keeping them within the method block makes more sense. This is like writing a class with private methods. 

But still if you suggest making a change. Will do it.